at_client/lib/src/service/sync_service_impl.dart

```diff
@@ -17,13 +17,17 @@ import 'package:at_lookup/at_lookup.dart';
 import 'package:at_persistence_secondary_server/at_persistence_secondary_server.dart';
 import 'package:at_utils/at_logger.dart';
 import 'package:at_utils/at_utils.dart';
+import 'package:uuid/uuid.dart';
 
 ///A [SyncService] object is used to ensure data in local secondary(e.g mobile device) and cloud secondary are in sync.
 class SyncServiceImpl implements SyncService, AtSignChangeListener {
-  bool _isSyncInProgress = false;
+  static const int _syncMaxRetryTime = 2,
+      _syncRequestThreshold = 3,
+      _syncRequestTriggerInSeconds = 30;
   late final AtClient _atClient;
   late final RemoteSecondary _remoteSecondary;
   late final NotificationServiceImpl _statsNotificationListener;
+  final _syncRequests = <SyncRequest>[];
```

Comment:
Make it a fixed size queue. i.e 5 th one will be replaced.

Reply:
done